### ISSUE TITLE
Allow for easily building a single runtime test

### ISSUE THREAD

**Issue by CoffeeFlux** [dotnet/runtime#44205]
As I understand it, you currently have to build the full runtime test suite before you can run any of them individually. If that is the case, it would be nice to be able to easily build a single project since the full suite takes a long time. Alternatively, if this is currently possible and easy, we should document how to do it.

**Comment by ViktorHofer** [comment]
cc @trylek

**Comment by jkoritzinsky** [comment]
If you use the `dotnet.cmd` or `dotnet.sh` scripts at the root, you can build a single runtime test via `dotnet msbuild src/tests/path/to/test.csproj` once you've generated the Core_Root layout. Then you can run it by directly running the `.cmd` or `.sh` scripts that the build generates.

**Comment by ViktorHofer** [comment]
It would be great if `dotnet test` invoked on the project file would do that for you. I understand that bootstrapping CORE_ROOT might not be possible yet as an incremental dependency but the building and invoking of run script should be feasible. Or is that already the case?

**Comment by jkoritzinsky** [comment]
We haven't hooked up `dotnet test` yet. My understanding is that we'd have to override the VSTest target since we don't use a test suite to run individual test projects.

**Comment by ViktorHofer** [comment]
Right, overriding the VSTest target is sufficient. That wasn't possible before NET5 because the CLI hid all output when using `dotnet test` but I fixed that some months ago.

**Comment by CoffeeFlux** [comment]
> once you've generated the Core_Root layout

Doesn't that require you to build the full suite? How would you do that individually?

**Comment by jkoritzinsky** [comment]
You can use the `generatelayoutonly` parameter to the `src/tests/build.cmd` or `src/tests/build.sh` scripts to just build Core_Root.

**Comment by CoffeeFlux** [comment]
I see. Ideally `dotnet test` would handle as much as possible, including building `Core_Root`, but beyond that I think we should update the documentation to make the full process clear. This would have saved me a lot of time even in its current form.

**Comment by trylek** [comment]
I use commands like dotnet msbuild src\tests\JIT\methodical\divrem\div\i4div_cs_do.csproj; you just need to remember to specify the build properties as necessary, e.g. /p:Configuration=Checked, /p:TargetArchitecture=x86; in short, it's somewhat annoying and worth thinking about simplifying. I plan to tackle the "dotnet test" switchover but I don't yet have any proof it will automatically make this easier.
